index.js

```diff
@@ -366,9 +368,23 @@ Archive.prototype.appendFile = function (filename, name, cb) {
       mode: st.mode,
       size: 0,
       link: null
-    }, {filename: filename}, cb)
+    }, {filename: filename}, function (err, entry) {
+      if (err) return cb(err)
+      if (!entry) return cb()
+      if (entry.type === 'file') self.readStats.files++
+      cb(err, entry)
+    })
 
-    if (ws) pump(fs.createReadStream(path.resolve(self.directory, filename)), ws)
+    var statsCounter = through(function (data, encoding, next) {
+      self.readStats.totalRead += data.length
+      this.push(data)
+      next()
+    })
+
+    if (ws) {
+      var readStream = fs.createReadStream(path.resolve(self.directory, filename))
+      pump(readStream, statsCounter, ws)
+    }
```

Comment:
@mafintosh that makes sense to return the stats for the single append rather than the whole archive. =)

I can get the appendFile to return the stats in the callback (the total append size is already there) but not while in progress on a single append (in the `through` stream). @maxogden, I may need your help here. The only way I could figure it out was to emit an event and use that in `Dat.addFiles`:

```javascript

Archive.appendFile = function(...) {
  ...
  var statsCounter = through(function (data, encoding, next) {
      this.push(data)
      self.emit('read-progress', data.length)
      next()
    })
  ...
}

Dat.addFiles = function(...) {
  var pack = this.drive.add('.')
  ...
  pack.on('read-progress', function(data) {
    stats.bytesRead += data
  })
  ...
}
```

Is there a better way monitor the read progress on a single file?

---

It seems like sometimes progress is tracked in both Hyperdrive and Dat, for different functions. A lot of the progress (read files, read bytes for finished files, etc.) can be either place, but some (e.g. reading progress on single file) can only be in hyperdrive. Should the goal be to keep as much progress stuff in dat except only hyperdrive can track things (e.g. `statsCounter`)? 

For instance, [this line](https://github.com/maxogden/dat/blob/master/index.js#L92) we moved from Dat to Hyperdrive but it is not quite clear to me which module it should be in.